File: advent-of-code-2024/day3/problem1.py
import re
import time

input_file = "./input.in"

with open(input_file, "r") as file:
    data = file.read()

valid_re = r"mul\(\d+,\d+\)"


def get_nums(st):
    numm = st[4:-1].split(",")
    return int(numm[0]), int(numm[1])


a = time.perf_counter()
all_matches = re.findall(valid_re, data)
num_ls = list(map(get_nums, all_matches))

summation = 0
for l in num_ls:
    summation += l[0] * l[1]

b = time.perf_counter()

print(summation)
print(b - a)

File: advent-of-code-2024/day3/problem1.jl
function func()# Input file
    input_file = "./input.in"

    # Read file content
    data = read(input_file, String)

    # Define the regex pattern
    valid_re = r"mul\((\d+),(\d+)\)"

    # Extract numbers using regex and calculate summation

    @time begin
        summ = sum(
            parse(Int, m.captures[1]) * parse(Int, m.captures[2]) for m in eachmatch(valid_re, data)
        )
    end
    print(summ)
end

func()